### PR TITLE
Further docstrings and examples for :class:`~.Mobject`

### PR DESCRIPTION
**Pull request by kolibril13**
As there are still a lot of undocumented functions, this pr will add some more docstings to mobject methods.
Follow up to #1206

### CODE REVIEW

**Comment by kolibril13** [comment]
I have no idea what could cause this error message in the docs:
````
Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/networkx/utils/decorators.py", line 396, in _random_state
    random_state_arg = args[random_state_index]
IndexError: tuple index out of range

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/sphinx/cmd/build.py", line 280, in build_main
    app.build(args.force_all, filenames)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/sphinx/application.py", line 348, in build
    self.builder.build_update()
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 297, in build_update
    self.build(to_build,
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 311, in build
    updated_docnames = set(self.read())
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 418, in read
    self._read_serial(docnames)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 439, in _read_serial
    self.read_doc(docname)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 479, in read_doc
    doctree = read_doc(self.app, self.env, self.env.doc2path(docname))
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/sphinx/io.py", line 221, in read_doc
    pub.publish()
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/core.py", line 217, in publish
    self.document = self.reader.read(self.source, self.parser,
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/sphinx/io.py", line 126, in read
    self.parse()
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/readers/__init__.py", line 77, in parse
    self.parser.parse(self.input, document)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/sphinx/parsers.py", line 102, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 170, in run
    results = StateMachineWS.run(self, input_lines, input_offset,
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/statemachine.py", line 241, in run
    context, next_state, result = self.check_line(
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/statemachine.py", line 459, in check_line
    return method(match, context, next_state)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2769, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 327, in section
    self.new_subsection(title, lineno, messages)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 393, in new_subsection
    newabsoffset = self.nested_parse(
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 281, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/statemachine.py", line 241, in run
    context, next_state, result = self.check_line(
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/statemachine.py", line 459, in check_line
    return method(match, context, next_state)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2344, in explicit_markup
    self.explicit_list(blank_finish)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2369, in explicit_list
    newline_offset, blank_finish = self.nested_list_parse(
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 318, in nested_list_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/statemachine.py", line 241, in run
    context, next_state, result = self.check_line(
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/statemachine.py", line 459, in check_line
    return method(match, context, next_state)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2647, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2354, in explicit_construct
    return method(self, expmatch)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2096, in directive
    return self.run_directive(
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2146, in run_directive
    result = directive_instance.run()
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/sphinx/ext/autodoc/directive.py", line 157, in run
    result = parse_generated_content(self.state, params.result, documenter)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/sphinx/ext/autodoc/directive.py", line 104, in parse_generated_content
    state.nested_parse(content, 0, node)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 281, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/statemachine.py", line 241, in run
    context, next_state, result = self.check_line(
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/statemachine.py", line 459, in check_line
    return method(match, context, next_state)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2342, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2354, in explicit_construct
    return method(self, expmatch)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2096, in directive
    return self.run_directive(
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2146, in run_directive
    result = directive_instance.run()
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/sphinx/domains/__init__.py", line 285, in run
    return super().run()
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/sphinx/directives/__init__.py", line 209, in run
    self.state.nested_parse(self.content, self.content_offset, contentnode)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 281, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/statemachine.py", line 241, in run
    context, next_state, result = self.check_line(
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/statemachine.py", line 459, in check_line
    return method(match, context, next_state)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2342, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2354, in explicit_construct
    return method(self, expmatch)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2096, in directive
    return self.run_directive(
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2146, in run_directive
    result = directive_instance.run()
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/checkouts/1232/docs/source/manim_directive.py", line 240, in run
    exec("\n".join(code), globals())
  File "<string>", line 20, in <module>
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/manim/scene/scene.py", line 183, in render
    self.construct()
  File "<string>", line 12, in construct
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/manim/mobject/graph.py", line 365, in __init__
    self._layout = _determine_graph_layout(
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/manim/mobject/graph.py", line 48, in _determine_graph_layout
    auto_layout = automatic_layouts[layout](
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/decorator.py", line 214, in fun
    return caller(func, *(extras + args), **kw)
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/networkx/utils/decorators.py", line 400, in _random_state
    raise nx.NetworkXError("random_state_index is incorrect") from e
networkx.exception.NetworkXError: random_state_index is incorrect

Exception occurred:
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1232/lib/python3.8/site-packages/networkx/utils/decorators.py", line 400, in _random_state
    raise nx.NetworkXError("random_state_index is incorrect") from e
networkx.exception.NetworkXError: random_state_index is incorrect
The full traceback has been saved in /tmp/sphinx-err-czae81b8.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```
any idea?

File: manim/mobject/mobject.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ class Mobject(Container):
     """
 
     def __init__(self, color=WHITE, name=None, dim=3, target=None, z_index=0, **kwargs):
-        self.color = Color(color)
+        self.color: Color = Color(color)
```
Comment (jsonvillanueva) [inline manim/mobject/mobject.py:57]: The typing hints should be done in the parameters one line above -- not in the definition/code

File: manim/mobject/mobject.py [hunk 2]
```diff
@@ -1839,12 +1841,15 @@ def match_coord(self, mobject: "Mobject", dim, direction=ORIGIN):
...
     def match_x(self, mobject: "Mobject", direction=ORIGIN):
+        """match x coord. to the x coord. of another :class:`~.Mobject`"""
         return self.match_coord(mobject, 0, direction)
 
     def match_y(self, mobject: "Mobject", direction=ORIGIN):
+        """match y coord. to the x coord. of another :class:`~.Mobject`"""
         return self.match_coord(mobject, 1, direction)
 
     def match_z(self, mobject: "Mobject", direction=ORIGIN):
+        """match z coord. to the x coord. of another :class:`~.Mobject`"""
```
Comment (jsonvillanueva) [inline manim/mobject/mobject.py:1852]: Capitalize M, ending periods

**Review by jsonvillanueva — COMMENTED** [review]
Besides the odd typing hint, everything else looks mostly fine... just a few capitalization/periods at the end needed.

File: manim/mobject/mobject.py (hunk 2 above)
Comment (kolibril13) [inline manim/mobject/mobject.py:1852]:
```suggestion
    def match_x(self, mobject: "Mobject", direction=ORIGIN):
        """Match x coord. to the x coord. of another :class:`~.Mobject`."""
        return self.match_coord(mobject, 0, direction)

    def Match_y(self, mobject: "Mobject", direction=ORIGIN):
        """match y coord. to the x coord. of another :class:`~.Mobject`."""
        return self.match_coord(mobject, 1, direction)

    def match_z(self, mobject: "Mobject", direction=ORIGIN):
        """Match z coord. to the x coord. of another :class:`~.Mobject`."""
```

File: manim/mobject/mobject.py (hunk 1 above)
Comment (kolibril13) [reply manim/mobject/mobject.py:57]: okay, I will remove this one then.
```suggestion
        self.color  = Color(color)
```

**Comment by kolibril13** [comment]
@ManimCommunity/docs , feel free to review!

File: manim/mobject/mobject.py [hunk 3]
```diff
@@ -2011,19 +2016,69 @@ def shuffle(self, recursive=False):
...
         return self.sort(*args, **kwargs)
 
     def shuffle_submobjects(self, *args, **kwargs):
+        """Shuffles the order of :attr:`submobjects`
+
+        Examples
+        --------
+
+        .. manim:: SuffleSumobjectsExample
+            :save_last_frame:
```
Comment (jsonvillanueva) [inline manim/mobject/mobject.py:2072]: Was this intended to be a video?
```suggestion
```

File: manim/mobject/mobject.py [hunk 4]
```diff
@@ -1879,12 +1881,15 @@ def match_coord(self, mobject: "Mobject", dim, direction=ORIGIN):
         )
 
     def match_x(self, mobject: "Mobject", direction=ORIGIN):
+        """Match x coord. to the x coord. of another :class:`~.Mobject`."""
         return self.match_coord(mobject, 0, direction)
 
-    def match_y(self, mobject: "Mobject", direction=ORIGIN):
+    def Match_y(self, mobject: "Mobject", direction=ORIGIN):
+        """match y coord. to the x coord. of another :class:`~.Mobject`."""
```
Comment (jsonvillanueva) [inline manim/mobject/mobject.py:1888]:
```suggestion
        """Match y coord. to the x coord. of another :class:`~.Mobject`."""
```

File: manim/mobject/mobject.py [hunk 5]
```diff
@@ -2011,19 +2016,69 @@ def shuffle(self, recursive=False):
...
+        .. manim:: ArrangeSumobjectsExample
+            :save_last_frame:
+
+            class ArrangeSumobjectsExample(Scene):
+                def construct(self):
+                    s= VGroup(*[Dot().shift(i*0.1*RIGHT*np.random.uniform(-1,1)+UP*np.random.uniform(-1,1)) for i in range(0,15)])
+                    s.shift(UP).set_color(BLUE)
+                    s2= s.copy().set_color(RED)
+                    s2.arrange_submobjects()
+                    s2.shift(DOWN)
```
Comment (jsonvillanueva) [inline manim/mobject/mobject.py:2056]: This example doesn't add/show it. You need to add s/s2 for image

File: manim/mobject/mobject.py [hunk 6]
```diff
@@ -2011,19 +2016,69 @@ def shuffle(self, recursive=False):
         random.shuffle(self.submobjects)
 
     def invert(self, recursive=False):
+        """Inverts the list of :attr:`submobjects`.
+
+        Examples
+        --------
+
+        .. manim:: InvertSumobjectsExample
+            :save_last_frame:
```
Comment (jsonvillanueva) [inline manim/mobject/mobject.py:2025]: Was this intended to be a video? It's unclear what this does from the image.

**Review by jsonvillanueva — CHANGES_REQUESTED** [review]

File: manim/mobject/mobject.py (hunk 6 above)
Comment (kolibril13) [reply manim/mobject/mobject.py:2025]: Oh yes, thanks, that was a copy paste  mistake

Comment (kolibril13) [inline manim/mobject/mobject.py:2025]:
```suggestion
```

File: manim/mobject/mobject.py (hunk 5 above)
Comment (kolibril13) [inline manim/mobject/mobject.py:2056]:
```suggestion
                    s2.shift(DOWN)
                    self.add(s,s2)
```

**Review by jsonvillanueva — APPROVED** [review]

**Pull request merged**